src/steps/active-directory/converters.ts

```diff
@@ -0,0 +1,227 @@
+import {
+  createIntegrationEntity,
+  createDirectRelationship,
+  Entity,
+  IntegrationInstance,
+  Relationship,
+  RelationshipClass,
+  createMappedRelationship,
+  getTime,
+} from '@jupiterone/integration-sdk-core';
+import { Organization, User } from '@microsoft/microsoft-graph-types';
+import { Finding, Machine, UserLogon } from '../../types';
+import { entities, MappedRelationships, TargetEntities } from './constants';
+
+export function createAccountEntity(instance: IntegrationInstance): Entity {
+  return createIntegrationEntity({
+    entityData: {
+      source: {},
+      assign: {
+        _class: entities.ACCOUNT._class,
+        _key: `${entities.ACCOUNT._type}-${instance.id}`,
+        _type: entities.ACCOUNT._type,
+        name: instance.name,
+        displayName: instance.name,
+      },
+    },
+  });
+}
+
+export function createAccountEntityWithOrganization(
+  instance: IntegrationInstance,
+  organization: Organization,
+  intuneConfig: {
+    mobileDeviceManagementAuthority?: string;
+    subscriptionState?: string;
+    intuneAccountID?: string;
+  },
+): Entity {
+  let defaultDomain: string | undefined;
+  const verifiedDomains = organization.verifiedDomains?.map((e) => {
+    if (e.isDefault) {
+      defaultDomain = e.name as string | undefined;
+    }
+    return e.name as string;
+  });
+
+  return createIntegrationEntity({
+    entityData: {
+      source: {
+        organization,
+        intuneConfig,
+      },
+      assign: {
+        _class: entities.ACCOUNT._class,
+        _key: `${entities.ACCOUNT._type}-${instance.id}`,
+        _type: entities.ACCOUNT._type,
+        id: organization.id,
+        name: organization.displayName,
+        displayName: instance.name,
+        organizationName: organization.displayName,
+        defaultDomain,
+        verifiedDomains,
+        intuneAccountId: intuneConfig?.intuneAccountID,
+        mobileDeviceManagementAuthority:
+          intuneConfig?.mobileDeviceManagementAuthority,
+        intuneSubscriptionState: intuneConfig?.subscriptionState,
+      },
+    },
+  });
+}
+
+export function createMachineEntity(data: Machine): Entity {
+  return createIntegrationEntity({
+    entityData: {
+      source: {}, // removed due to size
```

Comment:
Having seen this twice in the PR now, I'm wondering if there was a problem with setting `source: data`?